design-document.md

```diff
@@ -20,91 +41,151 @@
 1. Input validation protocol: The protocol executed by the client and
    aggregators in order to validate the client's input without leaking its value
    to the aggregators.
-1. Invalid input: An input for which the input validation protocol fails.  e.g.,
-   if the inputs are bit vectors, then `[2, 1, 0]` is invalid.
+1. Invalid input: An input for which the input validation protocol fails. For
+   example, if the input is meant to be a  bit vectors, then `[2, 1, 0]` is
+   invalid.
 1. Leader: A distinguished aggregator that coordinates input validation and data
-   aggregation.
+   collection.
 1. Output: A reduction over the inputs, for instance a statistical aggregation,
-   which is of interest to a collector.
+   which is of interest to a collector. This is the output of the aggregation
+   function.
 1. Output share: The share of an output emitted by an aggregator. Output shares
-   can be reassembled by the collector into the output.
+   can be reassembled by the leader into the final output.
+1. Prio v1: Mozilla's [Origin Telmetry proejct](https://blog.mozilla.org/security/2019/06/06/next-steps-in-privacy-preserving-telemetry-with-prio/).
+1. Prio v2: Contact tracing project by Apple, Google, and ISRG.
+1. Proof: A value generated by the client used by the aggregators to verify the
+   client's input.
+1. Proof share: A share of a proof, used by an aggregator during the
+   input-validation protocol.
+
+## Overview {#overview}
+
+The protocol is executed by a large set of clients and a small set of servers.
+We call the servers the *aggregators*. Each client holds an input. Given the set
+of inputs x[1], ..., x[n], the goal is to compute y = F(x[1], ..., x[n]) for
+some aggregation function F, while revealing nothing else about the inputs.
+
+### Secret sharing
+
+Prio achieves this goal using additive secret sharing. Rather than send its
+input in the clear, each client "splits" its input into a sequence of "shares"
+and sends a share to each of the aggregators. This secret-sharing procedure has
+two important properties: first, it is impossible to deduce the input given only
+a proper subset of the shares; and second, it allows the aggregators to compute
+the final output by first adding up their input shares locally, then combining
+the results to obtain the final output.
+
+Consider an illustrative example. Suppose there are three clients and two
+aggregators. Each client holds a positive integer, and our goal is to compute
+the sum of the inputs. Using an additive secret-sharing scheme, the first client
```

Comment:
This seems like an example where we actually have to invoke modular arithmetic, because obviously if x[,] are just integers then we know that x[1] < x[1,1] and x[1] < x[1,2].

Reply:
I took your suggestion above, which should make it clear.